packages/grafana-schema/src/schema/mudball.gen.ts

```diff
@@ -473,12 +479,38 @@ export enum BarGaugeDisplayMode {
   Lcd = 'lcd',
 }
 
+/**
+ * Table cell options. Each cell has a display mode
+ * along with sub options for that display mode. Sub options
+ * are grouped within the subOptions property by display mode.
+ */
+export interface TableCellOptions {
+  displayMode: TableCellDisplayMode;
+  subOptions: TableCellSubOptionGroup;
+}
+
+/**
+ * Each sub option group can have an arbitrary set of options.
+ */
+export type TableCellSubOptionGroup = Partial<Record<TableCellDisplayMode, TableCellSubOptions>>;
+
+/**
+ * Each cell will have it's own properties so
+ * the basic subOptions is simply empty.
+ */
+export interface TableCellSubOptions {};
+
+
+export const defaultTableCellOptions: Partial<TableCellOptions> = {
+  displayMode: TableCellDisplayMode.Auto,
+};
```

Comment:
I think it might make sense to type this using Typescript discriminator typing 

Example 
```ts

export interface LcdGaugeCellMode {
  displayMode: 'lcd-gauge';
  colorValue?: boolean; 
}

export interface ColoredBackgroundCellMode {
  displayMode: 'colored-background';
  colorMode: 'solid' | 'gradient'; 
}

export type TableCellDisplayMode = LcdGaugeCellMode | ColoredBackgroundCellMode; 

```

https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/templateVars.ts

Reply:
Thanks for the feedback Torkel! I'll see what I can do here 😄